src/services/espressoService.ts

```diff
@@ -0,0 +1,51 @@
+import { Machine } from "@prisma/client";
+import prisma from "../config/prisma";
+import CreateEspressoInput from "../interfaces/createEspressoInput";
+
+const getById = async (id: number) => {
+    return await prisma.espresso.findUnique({ where: { id } });
```

Comment:
Missing the includes which will return the machine, grinder and bean alongside the espresso.
[Link to docs](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#include-deeply-nested-relations)